src/bun.js/node/util/parse_args.zig

```diff
@@ -0,0 +1,687 @@
+    //
+    // Phase 0: parse the config object
+    //
+
+    const config = if (config_obj.isUndefinedOrNull()) null else config_obj;
+    if (config) |c| {
+        try validateObject(globalThis, c, "config", .{}, .{});
+    }
+
+    // Phase 0.A: Get and validate type of input args
+    var args: ArgsSlice = undefined;
+    const config_args_or_null = if (config) |c| c.getOwn(globalThis, "args") else null;
+    if (config_args_or_null) |config_args| {
+        try validateArray(globalThis, config_args, "args", .{}, null);
+        args = .{
+            .array = config_args,
+            .start = 0,
+            .end = @intCast(config_args.getLength(globalThis)),
+        };
+    } else {
+        args = try getDefaultArgs(globalThis);
+    }
+
+    // Phase 0.B: Parse and validate config
+
+    const config_strict: JSValue = (if (config) |c| c.getOwn(globalThis, "strict") else null) orelse JSValue.jsBoolean(true);
+    const config_allow_positionals: ?JSValue = if (config) |c| c.getOwn(globalThis, "allowPositionals") else null;
+    const config_return_tokens: JSValue = (if (config) |c| c.getOwn(globalThis, "tokens") else null) orelse JSValue.jsBoolean(false);
+    const config_options_obj: ?JSValue = if (config) |c| c.getOwn(globalThis, "options") else null;
+
+    try validateBoolean(globalThis, config_strict, "strict", .{});
+    const strict = config_strict.toBoolean();
+
+    var allow_positionals = !strict;
+    if (config_allow_positionals) |config_allow_positionals_value| {
+        try validateBoolean(globalThis, config_allow_positionals_value, "allowPositionals", .{});
+        allow_positionals = config_allow_positionals_value.toBoolean();
+    }
+
+    try validateBoolean(globalThis, config_return_tokens, "tokens", .{});
+    const return_tokens = config_return_tokens.toBoolean();
+
+    // Phase 0.C: Parse the options definitions
+
+    var options_defs_allocator = std.heap.stackFallback(2048, globalThis.allocator());
+    var option_defs = std.ArrayList(OptionDefinition).init(options_defs_allocator.get());
+    defer option_defs.deinit();
+
+    if (config_options_obj) |options_obj| {
+        try parseOptionDefinitions(globalThis, options_obj, &option_defs);
+    }
+
+    //
+    // Phase 1: tokenize the args string-array
+    //
+    log("Phase 1: tokenize args (args.len={d})", .{args.end - args.start});
+
+    var tokens_allocator = std.heap.stackFallback(32 * @sizeOf(Token), globalThis.allocator());
+    var tokens = try std.ArrayList(Token).initCapacity(tokens_allocator.get(), args.end - args.start);
+    defer tokens.deinit();
+
+    try tokenizeArgs(globalThis, args, option_defs.items, &tokens);
+
+    //
+    // Phase 2: process tokens into parsed option values and positionals
+    //
+    log("Phase 2: parse options from tokens (tokens.len={d})", .{tokens.items.len});
+
+    var result_values = JSValue.constructEmptyObject(globalThis, null, 0);
```

Comment:
use `createEmptyObject`

Reply:
I actually had that but changed it because it needs to be an object with null prototype instead of the Object prototype. Node even had some tests for this which were failing otherwise.